Lib/pathlib.py

```diff
@@ -878,7 +899,15 @@ def is_junction(self):
         """
         Whether this path is a junction.
         """
-        return os.path.isjunction(self)
+        import stat
+        try:
+            return self.lstat().st_reparse_tag == stat.IO_REPARSE_TAG_MOUNT_POINT
+        except OSError as e:
+            if not _ignore_error(e):
+                raise
+            return False
+        except (ValueError, AttributeError):
+            return False
```

Comment:
This implementation of `_PathBase.is_junction()` is new. It calls `self.lstat()` to get the stat result, and so it doesn't hit the underlying filesystem (unless `lstat()` does!). It would benefit from a review!

Reply:
Alternatively we could make `is_junction()` immediately raise `UnsupportedOperation`, but it would make it the only `is_*()` method that doesn't call through to `stat()`.

Reply:
Resolving this thread as I've made `_PathBase.is_junction()` immediately return false, which avoids all the Windows-specific stuff.